yt_dlp/downloader/hls.py

```diff
@@ -218,15 +225,19 @@ def is_ad_fragment_end(s):
                     decrypt_url = decrypt_info.get('URI')
                     decrypt_info = parse_m3u8_attributes(line[11:])
                     if decrypt_info['METHOD'] == 'AES-128':
-                        if 'IV' in decrypt_info:
+                        if external_aes_iv:
+                            decrypt_info['IV'] = external_aes_iv
+                        elif 'IV' in decrypt_info:
                             decrypt_info['IV'] = binascii.unhexlify(decrypt_info['IV'][2:].zfill(32))
-                        if not re.match(r'^https?://', decrypt_info['URI']):
-                            decrypt_info['URI'] = urllib.parse.urljoin(
-                                man_url, decrypt_info['URI'])
-                        if extra_query:
-                            decrypt_info['URI'] = update_url_query(decrypt_info['URI'], extra_query)
-                        if decrypt_url != decrypt_info['URI']:
-                            decrypt_info['KEY'] = None
+                        if external_aes_key:
+                            decrypt_info['KEY'] = external_aes_key
+                        else:
+                            if not re.match(r'^https?://', decrypt_info['URI']):
+                                decrypt_info['URI'] = urllib.parse.urljoin(man_url, decrypt_info['URI'])
```

Comment:
```suggestion
                            decrypt_info['URI'] = urljoin(man_url, decrypt_info['URI'])
```
The urljoin in utils should handle this

Reply:
`urllib.parse.urljoin` is used 2 other places in `hls.py`, should I change those as well?

Reply:
sure